### ISSUE TITLE
[Feature]: `cidr explain` 

### ISSUE THREAD

**Issue by bschaatsbergen** [bschaatsbergen/cidr#63]
We should consider adding a new subcommand: `cidr explain` to show relevant information about a CIDR range. Think of the explain subcommand as a feature to provide detailed insights into a given CIDR range.

## Proposal:

Introduce a new subcommand, `explain`, to `cidr`. When users run `cidr explain <CIDR>`, the tool should output information about the specified CIDR range. This information could for example include:

Network Address: The base IP address of the CIDR range.
Subnet Mask: The subnet mask associated with the CIDR range.
CIDR Notation: The CIDR notation for the range.
Number of Hosts: The total number of usable host addresses within the CIDR range.
Usable IP Range: The range of IP addresses that can be assigned to devices on the network.
Broadcast Address: The address used to broadcast messages to all devices on the network.

### Example Usage:

```
$ cidr explain 10.0.0.0/16

Network Address:   10.0.0.0
Subnet Mask:       255.255.0.0
CIDR Notation:      /16
Number of Hosts:    65,534
Usable IP Range:    10.0.0.1 to 10.0.255.254
Broadcast Address:  10.0.255.255
```

**Comment by bschaatsbergen** [comment]
What I implemented so far in #75:

![image](https://github.com/bschaatsbergen/cidr/assets/58337159/701f047c-0b23-4ec2-b9d3-3838837c3f5d)

**Comment by bschaatsbergen** [comment]
What I implemented so far in #75:
![image](https://github.com/bschaatsbergen/cidr/assets/58337159/34b3a9c5-cbb2-4e14-a383-80913041c310)
